recipe/meta.yaml

```
@@ -13,157 +14,236 @@
-    - nccl >=2.8,<3                         # [build_platform != target_platform]
-    - cutensor >=1.4,<2                     # [build_platform != target_platform]
-    - cusparselt ~=0.2.0.0                  # [build_platform != target_platform and (linux64 or aarch64 or win) and (cuda_compiler_version or "").startswith("11")]
-
   host:
     - python
-    - pip
-    - setuptools
-    - cython >=0.29.22,<3
-    - fastrlock >=0.5
-    - cuda-version {{ cuda_compiler_version }}
-    - nvtx-c  # [win64]
-    # TODO: clean up
-    {% if cuda_major >= 12 %}
-    - cuda-driver-dev  # [linux]
-    - cuda-cudart-dev
-    - cuda-nvrtc-dev
-    - cuda-nvtx-dev
-    - cuda-profiler-api
-    - cuda-cccl_{{ target_platform }}
-    - libcublas-dev
-    - libcufft-dev
-    - libcurand-dev
-    - libcusolver-dev
-    - libcusparse-dev
-    {% endif %}
-    # optional dependencies
-    # TODO: see https://github.com/conda-forge/cudnn-feedstock/issues/58
-    - cudnn >=8.0.*,<9      # [not ((aarch64 or ppc64le) and (cuda_compiler_version or "").startswith("12"))]
-    - nccl >=2.8,<3         # [linux]
-    - cutensor >=1.4,<2
-    - cusparselt ~=0.2.0.0  # [(linux64 or aarch64 or win) and (cuda_compiler_version or "").startswith("11")]
-
-  run:
-    - python
-    - {{ pin_compatible('fastrlock', max_pin='x.x') }}
-    - numpy >=1.21,<2
-    - cudatoolkit                                            # [(cuda_compiler_version or "").startswith("11")]
-    - cuda-version >={{ cuda_major }}.2,<{{ cuda_major+1 }}  # [(cuda_compiler_version or "").startswith("11")]
-    - cuda-version >={{ cuda_major }}.0,<{{ cuda_major+1 }}  # [(cuda_compiler_version or "").startswith("12")]
-
-  run_constrained:
-    # Only GLIBC_2.17 or older symbols present
-    - __glibc >=2.17  # [linux]
-    - scipy >=1.7,<2
-    - optuna >=3,<4
-    - __cuda >=11.2  # [(cuda_compiler_version or "").startswith("11")]
-    - __cuda >=12.0  # [(cuda_compiler_version or "").startswith("12")]
-    # TODO: Waiting for `aarch64` & `ppc64le` CUDA 12 support
-    - {{ pin_compatible('cudnn') }}                        # [not ((aarch64 or ppc64le) and (cuda_compiler_version or "").startswith("12"))]
-    - {{ pin_compatible('nccl') }}                         # [linux]
-    - {{ pin_compatible('cutensor', lower_bound='1.4') }}
-    - {{ pin_compatible('cusparselt', max_pin='x.x') }}    # [(linux64 or aarch64 or win) and (cuda_compiler_version or "").startswith("11")]
 
-test:
-  requires:
-    # need the libcuda stub for import test
-    - cuda-driver-dev  # [linux and (cuda_compiler_version or "").startswith("12")]
+outputs:
+  - name: cupy
+    version: {{ version }}
+    build:
+      number: {{ build_num }}
+      skip: true  # [(py < 39) or cuda_compiler_version not in ("11.8", "12.0")]
+    requirements:
+      host:
+        - python
+      run:
+        - python
+        - {{ pin_subpackage('cupy-core', max_pin='x.x.x') }}
```

Think we may want an exact match here

```suggestion
        - {{ pin_subpackage('cupy-core', exact=True) }}
```

---

done in commit [230b504](https://github.com/conda-forge/cupy-feedstock/pull/241/commits/230b504c508249f94cfc23eb3729e9631b0662f4)